docs/core/extensions/options-library-authors.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Options pattern guidance for .NET library authors
+author: IEvangelist
+description: Learn how to expose the options pattern as a library author in .NET.
+ms.author: dapine
+ms.date: 01/20/2021
+---
+
+# Options pattern guidance for .NET library authors
+
+With the help of dependency injection, registering your services and their corresponding configurations can make use of the *options pattern*. The options pattern enables consumers of your library (and your services) to require instances of <xref:Microsoft.Extensions.Options.IOptions%601> where `TOptions` is your options class. As a .NET library author, you'll learn general guidance on how to correctly expose the options pattern to consumers of your library. There are various ways to achieve the same thing, and several considerations to make.
+
+## Naming conventions
+
+By convention, extension methods responsible for registering services are named `Add{Service}`, where `{Service}` is a meaningful and descriptive name. Depending on the package, the registration of services may be accompanied by `Use{Service}` extension methods. The `Use{Service}` extension methods are commonplace in [ASP.NET Core](/aspnet).
+
+✔️ CONSIDER names that disambiguate your service from other offerings.
+
+❌ DO NOT use names that are already part of the .NET ecosystem from official Microsoft packages.
+
+✔️ CONSIDER naming static classes that expose extension methods as `{Type}Extensions`, where `{Type}` is the type that you're extending.
+
+### Namespace guidance
+
+Microsoft packages make use of the `Microsoft.Extensions.DependencyInjection` namespace to unify the registration of various service offerings.
+
+✔️ CONSIDER a namespace that clearly identifies your package offering.
+
+❌ DO NOT use the `Microsoft.Extensions.DependencyInjection` namespace for non-official Microsoft packages.
+
+## Parameterless
+
+If your service can work with minimal or no explicit configuration, consider a parameterless extension method.
+
+:::code language="csharp" source="snippets/configuration/options-noparams/ServiceCollectionExtensions.cs" highlight="10-14":::
+
+In the preceding code, the `AddMyLibraryService`:
+
+- Extends an instance of <xref:Microsoft.Extensions.DependencyInjection.IServiceCollection>
+- Calls <xref:Microsoft.Extensions.DependencyInjection.OptionsServiceCollectionExtensions.AddOptions%60%601(Microsoft.Extensions.DependencyInjection.IServiceCollection)?displayProperty=nameWithType> with the type parameter of `LibraryOptions`
+- Chains a call to <xref:Microsoft.Extensions.Options.OptionsBuilder%601.Configure%2A>, which specifies the default option values
+
+As the library author, specifying default values is up to you.
+
+> [!NOTE]
```

Comment:
I would move this to the next section and just mention that you may choose to read from configuration in the default overloads (see next section for more details).

Reply:
So rather than having an alert, styling as a callout - mention it in the next section?

Reply:
Yep it belongs with the configuration section.